packages/effect/src/Effect.ts

```diff
@@ -7884,12 +7884,12 @@ export const Do: Effect<{}> = effect.Do
  */
 export const bind: {
   <N extends string, A extends object, B, E2, R2>(
-    name: Exclude<N, keyof A>,
+    name: Exclude<N, effect.ForbiddenKeys<A>>,
```

Comment:
You'll need a publicly exported type, outside of `internal` otherwise this type won't exist in the compiled dts

Reply:
Thanks for the fast feedback @mikearnaldi ! I've got this in the works with corresponding updates for other uses of internal doNotation (Either/Option/etc)

Should have another commit up tomorrow

Reply:
All set!